docs: add missing podman task config options (#25465)
hashicorp/nomad: website/content/plugins/drivers/podman.mdx

## Task Configuration

> - `apparmor_profile` - (Optional) Name of a apparmor profile to be used instead of
>   the default profile. The special value unconfined disables apparmor for this container.

Reviewer: ```suggestion
- `apparmor_profile` - (Optional) Name of an AppArmor profile to use instead of
  the default profile. The special value `unconfined` disables AppArmor for this container.
```

Style: active voice (to be used -> to use)

and capitalized the product name

  }
  ```

> - `auth_soft_fail` - (Optional) Ignores errors returned from auth backend so
>   fallbacks can be tried.

Reviewer: ```suggestion
- `auth_soft_fail` - (Optional) Ignore errors returned from auth backend so Podman can fall back to a different auth method.
```

We are trying to eliminate passive voice (fallbacks can be tried) and use active voice instead. I'm not really sure about Podman so guessed at the functionality based on some searching.

  }
  ```

> - `cpu_hard_limit` (Optional) `true` or `false`. Use hard CPU limiting instead of
>   soft limiting. By default this is `false` which means soft limiting is used and
>   containers are able to burst above their CPU limit when there is idle capacity.

Reviewer: ```suggestion
- `cpu_hard_limit` (Optional) `true` or `false`. Use hard CPU limiting instead
  of soft limiting. By default this is `false`, which means Podman uses soft
  limiting so that containers are able to burst above their CPU limit when there
  is idle capacity.
```

Style: active voice and a bit of clarification

> - `cpu_cfs_period` - (Optional) Set the CPU persiod for the Completely Fair Scheduler (CFS),
>   which is a duration in microseconds. See podman --cpu-period for details.

Reviewer: ```suggestion
- `cpu_cfs_period` - (Optional) Set the CPU period for the [Completely Fair
  Scheduler (CFS)](https://docs.kernel.org/scheduler/sched-design-CFS.html),
  which is a duration in microseconds. Refer to [`podman run
  --cpu-period`](https://docs.podman.io/en/latest/markdown/podman-run.1.html#cpu-period-limit)
  for details.
```

I added some links

  }
  ```

> - `pids_limit` - (Optional) An integer value that specified the pid limit for

Reviewer: ```suggestion
- `pids_limit` - (Optional) An integer value that specified the PID limit for
```

  }
  ```

> - `security_opt` - (Optional) A list of security-related options that are set
>   in the container.

Reviewer: ```suggestion
- `security_opt` - (Optional) A list of security-related options that Podman sets
  in the container.
```

Style: use active voice - and I guessed that Podman sets these options in the container after reading some of the Podman docs.

  ```hcl
  config {
    security_opt = [
      "no-new-privileges"
    ]

  }
  ```

> - `selinux_opts` - (Optional) A list of prciess labels the container will use.

Reviewer: ```suggestion
- `selinux_opts` - (Optional) A list of process labels that the container uses.
```

style: present tense

  ```hcl
  config {
    selinux_opts = [
      "type:my_container.process"
    ]
  }
  ```

> - `shm_size` - (Optional) Set the size of `/dev/shm`. See Podman --shm-size
>   for more details.

Reviewer: ```suggestion
- `shm_size` - (Optional) Set the size of `/dev/shm`. Refer to `podman run
  --shm-size](https://docs.podman.io/en/latest/markdown/podman-run.1.html#shm-size-number-unit)
  for more details.
```

added link

  }
  ```

- `security_opt` - (Optional) A list of security-related options that Podman sets
  in the container.

  ```hcl
  config {
    security_opt = [
      "no-new-privileges"
    ]

  }
  ```

- `selinux_opts` - (Optional) A list of process labels that the container uses.

  ```hcl
  config {
    selinux_opts = [
      "type:my_container.process"
    ]
  }
  ```

> - `shm_size` - (Optional) Set the size of `/dev/shm`. Refer to `podman run
>   --shm-size](https://docs.podman.io/en/latest/markdown/podman-run.1.html#shm-size-number-unit)
>   for more details.

Reviewer: ```suggestion
- `shm_size` - (Optional) Set the size of `/dev/shm`. Refer to [`podman run
  --shm-size](https://docs.podman.io/en/latest/markdown/podman-run.1.html#shm-size-number-unit)
  for more details.
```

OOPS forgot the opening [
